vtysh/vtysh.c

```diff
@@ -3157,6 +3183,10 @@ void vtysh_init_vty(void)
 	install_element(BGP_EVPN_VNI_NODE, &vtysh_quit_bgpd_cmd);
 	install_element(BGP_IPV6L_NODE, &vtysh_exit_bgpd_cmd);
 	install_element(BGP_IPV6L_NODE, &vtysh_quit_bgpd_cmd);
+	install_element(BGP_VPNPOLICY_IPV4_NODE, &vtysh_exit_bgpd_cmd);
+	install_element(BGP_VPNPOLICY_IPV4_NODE, &vtysh_quit_bgpd_cmd);
+	install_element(BGP_VPNPOLICY_IPV6_NODE, &vtysh_exit_bgpd_cmd);
+	install_element(BGP_VPNPOLICY_IPV6_NODE, &vtysh_quit_bgpd_cmd);
```

Comment:
Please add the following lines:
```
install_element(BGP_VPNPOLICY_IPV4_NODE, &vtysh_end_all_cmd);
install_element(BGP_VPNPOLICY_IPV6_NODE, &vtysh_end_all_cmd);
```
Without these, I believe typing 'end' while in either of these nodes will cause an infinite recursion leading to a stack overflow and subsequent vtysh crash.

Reply:
Thanks for catching.